#! /usr/bin/env python

PACKAGE='depthimage_to_laserscan'

from dynamic_reconfigure.parameter_generator_catkin import *

from math import pi

gen = ParameterGenerator()
#       Name                    Type      Reconfiguration level             Description                                                            Default    Min   Max
gen.add("scan_height",          int_t,    0,                                "Height of the laser band (in pixels).",                            1,      1,   500)
gen.add("scan_offset",          double_t, 0,                                "Row set as the center of laserscan.",                              0.500,  0.0, 1.0)
gen.add("scan_time",            double_t, 0,                                "Time for the entire scan sweep.",                                  0.033,  0.0, 1.0)
gen.add("range_min",            double_t, 0,                                "Minimum reported range (in meters).",                              0.45,   0.0, 10.0)
gen.add("range_max",            double_t, 0,                                "Maximum reported range (in meters).",                              10.0,   0.0, 10.0)
gen.add("output_frame_id",      str_t,    0,                                "Output frame_id for the laserscan.",   "camera_depth_frame")

exit(gen.generate(PACKAGE, "depthimage_to_laserscan", "Depth"))